Fix activation of invited users during sharing

oliverguenther opened the pull request:
https://community.openproject.org/work_packages/51262

oliverguenther committed:
Fix activation of invited users during sharing

https://community.openproject.org/work_packages/51262

klaustopher reviewed (approved):
Looks good to me. Checked out locally and does what it says

oliverguenther committed:
Do not set a back_url when logging in through omniauth

oliverguenther committed:
Only set the omniauth back_url during direct login

Otherwise, users will be redirected to weird places like login or account/lost_password

oliverguenther commented on app/controllers/concerns/accounts/omniauth_login.rb:
```diff
@@ -47,8 +47,12 @@ module Accounts::OmniauthLogin
+    # in the original requested URL after logging in
+    if omniauth_direct_login?
+      params[:back_url] = request.env['omniauth.origin']
+    end
```
FYI @machisuji @ulferts @klaustopher  This change is necessary to prevent users from being redirected back to the back_url now that redirect_after_login uses the back_url presence. We need it for the sharing as an invited user will have this flow:

1. Click /account/activate?token=foo&back_url=/work_packages/bla
2. Do the registration form or whatever flow, save the back_url in the session of `AccountController#activate_invited`
3. Come back from authentication stages, expect to be redirected to work packages page

Could you double check this change please?

machisuji reviewed (approved):
👍
